docs/root/start/quick-start/securing.rst

```diff
@@ -0,0 +1,198 @@
+.. _start_quick_start_securing:
+
+Securing Envoy
+==============
+
+Envoy provides a number of features to secure traffic in and out of your network, and
+between proxies and services within your network.
+
+Transport Layer Security (``TLS``) can be used to secure all types of ``HTTP`` traffic, including ``WebSockets``.
+
+Envoy also has support for transmitting and receiving generic ``TCP`` traffic over ``TLS``.
+
+.. warning::
+
+   The following guide takes you through individual aspects of securing traffic.
+
+   To secure traffic over a network that is untrusted, you are strongly advised to make
+   use of :ref:`SNI <start_quick_start_securing_sni>` *and* :ref:`mTLS <start_quick_start_securing_mtls>`
```

Comment:
re: SNI, I would rephrase this, as I don't think it's required to use SNI. In some sense, SNI is just a routing concept (setting aside TLS1.3 encrypted SNI which breaks all of this).

re: mTLS, I might rephrase this slightly, as the way this is written will be contentious with some people. There is a set of people that think mTLS is too complicated and per-request (signing, etc. like JWT) is a better way to handle mutual authentication. I think mTLS is one way of doing this, but it would be good to rephrase slightly and potentially link to some of the per-request options such as JWT/Oauth/etc.

Reply:
so re SNI, my understanding is that it prevents a type of spoofing - ie valid certs for wrong server (maybe this is wrong).

ill remove this/related advice and try to clarify what is written about it...

re mTLS - got it (i think) - ill make the warning about mutual authentication and encryption, and suggest this as a way to address those problems

Reply:
> so re SNI, my understanding is that it prevents a type of spoofing - ie valid certs for wrong server (maybe this is wrong).

yep, from further testing, this is wrong.

sni "allows" multiple domains to be served at same ip with different certs. I guess browsers just automatically use the dns name here.

From envoy configuration pov i guess the advice is still that its best to put the dns of upstream here

Im less clear how important it is to include in sec section, but ill try and reword and we can figure what/whether to include